js/coinsbit.js

```diff
@@ -0,0 +1,708 @@
+                }
+                const response = await this[method] (this.deepExtend (request, params));
+                // {
+                //     success: true,
+                //     message: "",
+                //     result: [
+                //         {
+                //             id: 99601,
+                //             type: "buy",
+                //             time: 1640108002.455,
+                //             amount: "323144.05252658",
+                //             price: "0.008",
+                //             total: "2585.15242021"
+                //         },
+                //         {
+                //             id: 99544,
+                //             type: "buy",
+                //             time: 1640107990.112,
+                //             amount: "967.91335",
+                //             price: "0.00794371",
+                //             total: "7.68882295"
+                //         },
+                data = this.safeValue (response, 'result', []);
+            } else if (method === 'spotPublicGetHistoryResult') {
+                if (!('since' in params)) {
+                    throw new ArgumentsRequired (this.id + " fetchTrades() requires 'since' id in params");
+                }
+                data = this[method] (this.deepExtend (request, params));
+                // [
+                //     {
+                //         tid: 99601,
+                //         type: "buy",
+                //         date: 1640108002,
+                //         amount: "323144.05252658",
+                //         price: "0.008",
+                //         total: "2585.15242021"
+                //     },
+                //     {
+                //         tid: 99544,
+                //         type: "buy",
+                //         date: 1640107990,
+                //         amount: "967.91335",
+                //         price: "0.00794371",
+                //         total: "7.68882295"
+                //     },
+            }
+        }
+        return this.parseTrades (data, market, since, limit);
+    }
+
+    parseTrade (trade, market = undefined) {
+        //         /history
+        //         {
+        //             id: 99601,
+        //             type: "buy",
+        //             time: 1640108002.455,
+        //             amount: "323144.05252658",
+        //             price: "0.008",
+        //             total: "2585.15242021"
+        //         },
+
+        //         /history/result
+        //         {
+        //             tid: 99601,
+        //             type: "buy",
+        //             date: 1640108002,
+        //             amount: "323144.05252658",
+        //             price: "0.008",
+        //             total: "2585.15242021"
+        //         },
+        const symbol = market['symbol'];
+        const id = this.safeString2 (trade, 'id', this.safeString2 (trade, 'tid'));
```

Comment:
This is a wrong usage of safeString2, should be:
```JavaScript
const id = this.safeString2 (trade, 'id', 'tid');
```
In general, `safeString` is for one key, `safeString2` is for 2 keys.